src/main/java/com/libraryproject/librarysystem/book/BookController.java

```diff
@@ -0,0 +1,30 @@
+package com.libraryproject.librarysystem.book;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/books")
+public class BookController {
+    private final BookService bookService;
+
+    public BookController(BookService bookService) {
+        this.bookService = bookService;
+    }
+
+//    @GetMapping
+//    public ResponseEntity<Page<Book>> getAllBooks(@RequestParam(required = false) Integer page){
+//        int pageNumber = (page != null && page >= 0) ? page : 0;
+//        return ResponseEntity.ok(bookService.findAllBooksWithPagination(pageNumber));
+//    }
+
+    @GetMapping
+    public ResponseEntity<List<Book>> getAllBooks(){
```

Comment:
zwracanie encji z połączeniami onetomany itp jest bardzo trudne, po to własnie używamy mappera i dto, ale wiem że to nie miało być rozwiązanie docelowe :)

Reply:
Po zmienieniu na BookResponseDto wszystko działa, dzięki za komentarz. Dodałam też endpointa zwracającego page. Czy zostawić tylko jego?

Reply:
pozostaw oba. paginacja jest super przy większej ilosci danych, póki co dla ułatwienia możemy uzywać zwykłego endpointu